src/main/java/ch/wisv/areafiftylan/service/SeatServiceImpl.java

```diff
@@ -1,43 +1,124 @@
 package ch.wisv.areafiftylan.service;
 
+import ch.wisv.areafiftylan.dto.SeatGroupDTO;
+import ch.wisv.areafiftylan.dto.SeatmapResponse;
+import ch.wisv.areafiftylan.exception.SeatNotFoundException;
+import ch.wisv.areafiftylan.exception.TeamNotFoundException;
 import ch.wisv.areafiftylan.model.Seat;
+import ch.wisv.areafiftylan.model.Team;
+import ch.wisv.areafiftylan.model.Ticket;
 import ch.wisv.areafiftylan.model.User;
-import ch.wisv.areafiftylan.model.util.Coordinate;
-import ch.wisv.areafiftylan.service.repository.SeatRepository;
+import ch.wisv.areafiftylan.service.repository.SeatRespository;
+import ch.wisv.areafiftylan.service.repository.TicketRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-import java.util.List;
+import java.util.*;
 
 @Service
 public class SeatServiceImpl implements SeatService {
 
-    SeatRepository seatRepository;
+    SeatRespository seatRespository;
+    TicketRepository ticketRepository;
+    TeamService teamService;
 
     @Autowired
-    public SeatServiceImpl(SeatRepository seatRepository) {
-        this.seatRepository = seatRepository;
+    public SeatServiceImpl(SeatRespository seatRespository, TicketRepository ticketRepository,
+                           TeamService teamService) {
+        this.seatRespository = seatRespository;
+        this.ticketRepository = ticketRepository;
+        this.teamService = teamService;
     }
 
     @Override
-    public Seat getSeatByUser(User user) {
-        return this.seatRepository.findByUser(user);
+    public Seat getSeatByUsername(String username) {
+        return seatRespository.findByTicketOwnerUsername(username)
+                .orElseThrow(() -> new SeatNotFoundException("User " + username + " doesn't have a seat"));
     }
 
     @Override
-    public List<Seat> getAllSeats() {
-        return seatRepository.findAll();
+    public SeatmapResponse getAllSeats() {
+        List<Seat> all = seatRespository.findAll();
+
+        Map<String, List<Seat>> seatMapResponse = new HashMap<>();
+
+        for (Seat seat : all) {
```

Comment:
I think a solution with streams is easier to grasp. Something along the lines of

```java
all.stream().map(Seat::getSeatGroup).forEach(g -> seatMapResponse.put(g, new ArrayList<>()));
all.stream().forEach(s -> seatMapResponse.get(s.getSeatGroup).add(s);
```